qa/tasks/keystone.py

```diff
@@ -0,0 +1,405 @@
+"""
+Deploy and configure Keystone for Teuthology
+"""
+from cStringIO import StringIO
+from configobj import ConfigObj
+import argparse
+import base64
+import contextlib
+import logging
+import os
+import random
+import string
+import sys
+
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology import safepath
+from teuthology.config import config as teuth_config
+from teuthology.orchestra import run
+from teuthology.orchestra.connection import split_user
+from teuthology.packaging import install_package
+from teuthology.packaging import remove_package
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def install_packages(ctx, config):
+    """
+    Download the packaged dependencies of Keystone.
+    Remove install packages upon exit.
+
+    The context passed in should be identical to the context
+    passed in to the main task.
+    """
+    assert isinstance(config, dict)
+    log.info('Installing packages for Keystone...')
+
+    deps = [
+        'libffi-dev',
+        'libssl-dev',
+        'libldap2-dev',
+        'libsasl2-dev',
+        'python-openstackclient'
+    ]
+    for (client, _) in config.items():
+        (remote,) = ctx.cluster.only(client).remotes.iterkeys()
+        for pkgname in deps:
+            install_package(pkgname, remote)
+    try:
+        yield
+    finally:
+        log.info('Removing packaged dependencies of Keystone...')
+
+        for (client, _) in config.items():
+            (remote,) = ctx.cluster.only(client).remotes.iterkeys()
+            for pkgname in deps:
+                remove_package(pkgname, remote)
+
+@contextlib.contextmanager
+def download(ctx, config):
+    """
+    Download the Keystone from github.
+    Remove downloaded file upon exit.
+
+    The context passed in should be identical to the context
+    passed in to the main task.
+    """
+    assert isinstance(config, dict)
+    log.info('Downloading keystone...')
+    testdir = teuthology.get_testdir(ctx)
+    s3_branches = [ 'giant', 'firefly', 'firefly-original', 'hammer' ]
```

Comment:
Removed the extra logic. At the moment it resembles the idea proposed in #16219.